modules/rating/src/main/java/glicko2/RatingCalculator.java

```diff
@@ -61,14 +64,34 @@ public RatingCalculator(
    * @param results
    */
   public void updateRatings(RatingPeriodResults results) {
+    updateRatings(results, null, 0);
+  }
+
+  /**
+   * <p>Run through all players within a resultset and calculate their new ratings.</p>
+   * <p>Players within the resultset who did not compete during the rating period
+   * will have see their deviation increase (in line with Prof Glickman's paper).</p>
+   * <p>Note that this method will clear the results held in the association resultset.</p>
+   *
+   * @param results
+   * @param ratingPeriodEndDate
+   * @param ratingPeriodLengthMillis
+   */
+  public void updateRatings(RatingPeriodResults results, DateTime ratingPeriodEndDate, double ratingPeriodDays) {
     for ( Rating player : results.getParticipants() ) {
+      double elapsedRatingPeriods = 1;
+      if ( ratingPeriodEndDate != null && ratingPeriodDays != 0 && player.getLatest() != null ) {
+        Duration interval = new Duration(player.getLatest(), ratingPeriodEndDate);
+        elapsedRatingPeriods = interval.getMillis() / (ratingPeriodDays * MILLIS_PER_DAY);
```

Comment:
nit: can use `getStandardDays()` instead of a MILLIS_PER_DAY constant

Reply:
That returns an integer, I want a floating point. I couldn't find an equivalent.

Reply:
then just store `millisPerRatingPeriod`

Reply:
Makes sense. Now that I think about it, `ratingPeriodDays` should probably be a constructor parameter on `RatingCalculator`. Then it can precalculate `periodsPerMilli` and still use the simpler constant